api/core/listen_result.go

```diff
@@ -37,39 +37,38 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
 			})
 		}
 	}
-	return
+	return nil
 }
 
-func validateTaskKey(service *service.Service, taskKey string) (err error) {
+func validateTaskKey(service *service.Service, taskKey string) error {
 	if taskKey == "" || taskKey == "*" {
-		return
+		return nil
 	}
 	_, ok := service.Tasks[taskKey]
 	if ok {
-		return
+		return nil
 	}
-	err = errors.New("Task '" + taskKey + "' doesn't exist in this service")
-	return
+	return fmt.Errorf("Task '%q' doesn't exist in this service", taskKey)
 }
 
-func validateOutputKey(service *service.Service, taskKey string, outputFilter string) (err error) {
+func validateOutputKey(service *service.Service, taskKey string, outputFilter string) error {
 	if outputFilter == "" || outputFilter == "*" {
-		return
+		return nil
 	}
 	if taskKey == "" {
-		err = errors.New("Cannot filter output without specifying a task")
-		return
+		return fmt.Errorf("Cannot filter output without specifying a task")
 	}
 	task, ok := service.Tasks[taskKey]
 	if !ok {
-		err = errors.New("Task '" + taskKey + "' doesn't exist in this service")
-		return
+		return fmt.Errorf("Task '%q' doesn't exist in this service", taskKey)
 	}
+
+	var err error
 	_, ok = task.Outputs[outputFilter]
 	if !ok {
-		err = errors.New("Output '" + outputFilter + "' doesn't exist in the task '" + taskKey + "' of this service")
+		err = fmt.Errorf("Output %q doesn't exist in the task %q of this service", outputFilter, taskKey)
```

Comment:
return err here directly

Reply:
I remove the `if taskKey == ""` test and return the error `Output %q doesn't exist in the....` directly